internal/ecrconsumer/image_values_test.go

```diff
@@ -302,12 +302,31 @@ func TestWithImage(t *testing.T) {
 		t.Fatal("no matching image found")
 	}
 
-	imgVals := table(outputRls.Spec.Values, path)
+	outputImg := outputRls.Spec.Values["image"].(map[string]interface{})
+	assert.Equal(t, expectedImg.Tag, outputImg["tag"].(string))
 
-	outputImage, err := parseImage(imgVals["repository"].(string), imgVals["tag"].(string))
-	assert.NoError(t, err)
+	// Test No Matching Image Case
+	rls = v1alpha1.HelmRelease{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "HelmRelease",
+			APIVersion: "apiVersion: helmreleases.k8s.wattpad.com/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "example-microservice",
+		},
+		Spec: v1alpha1.HelmReleaseSpec{
+			ReleaseName: "example-release",
+			Chart: v1alpha1.ChartSpec{
+				Repository: "wattpad.s3.amazonaws.com/helm-charts",
+				Path:       "microservice",
+				Revision:   "HEAD",
+			},
+			Values: map[string]interface{}{},
```

Comment:
The `Values` shouldn't be empty, it defeats the point of the test case which is that non-matching image values shouldn't be changed. We don't really test that if there's no image values at all. If we want a 3rd case for no values, that could be a separate test but I don't think it's necessary.

Also I would avoid duplicating the big data (the HelmRelease struct) and rather duplicate the small data (desired Image) instead. We could use the same release struct in both cases and have 2 images, one of which wouldn't match the release.